static/app/views/dashboards/widgetBuilder/components/newWidgetBuilder.spec.tsx

```diff
@@ -69,6 +76,46 @@ describe('NewWidgetBuiler', function () {
     expect(await screen.findByRole('button', {name: '14D'})).toBeInTheDocument();
     expect(await screen.findByRole('button', {name: 'All Releases'})).toBeInTheDocument();
 
+    expect(await screen.findByPlaceholderText('Name')).toBeInTheDocument();
+    expect(await screen.findByTestId('add-description')).toBeInTheDocument();
+
     expect(await screen.findByText('TEST WIDGET')).toBeInTheDocument();
   });
+
+  it('edits name and description', async function () {
+    const mockNavigate = jest.fn();
+    mockUseNavigate.mockReturnValue(mockNavigate);
+
+    render(<DevWidgetBuilder isOpen onClose={onCloseMock} />, {
+      router,
+      organization,
+    });
+
+    await userEvent.type(await screen.findByPlaceholderText('Name'), 'some name');
+    // use this because navigate is called many times while the field is being updated
+    const lastTitleNavigateCall =
+      mockNavigate.mock.calls[mockNavigate.mock.calls.length - 1][0];
+    expect(lastTitleNavigateCall).toEqual(
+      expect.objectContaining({
+        ...router.location,
+        query: expect.objectContaining({title: 'some name'}),
+      })
+    );
+
+    await userEvent.click(await screen.findByTestId('add-description'));
```

Comment:
Typically we only want to resort to a test ID if we absolutely have to. RTL's philosophy is that you're supposed to test what your users see and a test ID isn't visible, but the text `+ Add Widget Description` is user-facing